src/libs/actions/Report.js

```diff
@@ -495,6 +495,85 @@ function fetchChatReports() {
         });
 }
 
+/**
+ * Get a simplified version of an IOU report
+ *
+ * @param {Number} reportID
+ * @param {Object} reportData
+ * @param {Number} reportData.transactionID
+ * @param {Number} reportData.amount
+ * @param {String} reportData.currency
+ * @param {String} reportData.created
+ * @param {String} reportData.comment
+ * @param {Object[]} reportData.transactionList
+ * @param {String} reportData.ownerEmail
+ * @param {String} reportData.managerEmail
+ * @returns {Object}
+ */
+function getSimplifiedIOUReport(reportID, reportData) {
+    const transactions = _.map(reportData.transactionList, transaction => ({
+        transactionID: transaction.transactionID,
+        amount: transaction.amount,
+        currency: transaction.currency,
+        created: transaction.created,
+        comment: transaction.comment,
+    }));
+
+    return {
+        reportID,
+        ownerEmail: reportData.ownerEmail,
+        managerEmail: reportData.managerEmail,
+        currency: reportData.currency,
+        transactions,
+    };
+}
+
+/**
+ * Fetches the updated data for an IOU Report and updates the IOU collection in ONYX
+ *
+ * @param {Number} reportID
+ * @param {Object[]} reportHistory
+ */
+function updateIOUReportData(reportID, reportHistory) {
+    const containsIOUAction = _.any(reportHistory, action => action.actionName === 'IOU');
+
+    // If there aren't any IOU actions, we don't need to fetch any additional data
+    if (!containsIOUAction) {
+        return;
+    }
+
+    const otherParticipants = _.chain(reportHistory)
+        .pluck('actorEmail')
+        .unique()
+        .without(currentUserEmail)
+        .value();
+
+    // If we have more than one participant, this is not an IOU
+    if (otherParticipants.length > 1) {
+        return;
+    }
+
+    // If we have an IOU action, get the IOU reportID
+    let iouReportID = 0;
+    API.GetIOUReport({
+        debtorEmail: otherParticipants[0],
+    }).then((response) => {
+        iouReportID = response.reportID;
+
+        Log.info('[Report] Fetching IOU report data', true, iouReportID);
+        return API.Get({
```

Comment:
Yeah, it's kind of a shame we have to make dual requests. Since this is a new feature, I say we take the time to do it properly and just have this data returned from `GetIOUReport`.

Reply:
@Julesssss do you have any reason why we shouldn't update `GetIOUReport` to just return the data from `GetReportStuff` on the API side? This sounds like a good improvement to me

Reply:
Nope, that seems like a good improvement to me too, I recall DD discussions about this and other improvements... I believe the counter argument was that it was an early optimisation that could come later.

CC'ing @iwiznia to get your thoughts here

Reply:
Let's not block on this, yes, we are doing 2 requests but that's not that bad. We can optimize this later.